packages/teleport/src/Databases/AddDatabase/AddDatabase.tsx

```diff
@@ -75,10 +75,19 @@ export function AddDatabase({
     Option<DatabaseInfo>
   >(dbOptions[0]);
 
-  const connectCmd =
-    authType === 'sso'
-      ? `tsh login --proxy=${host}`
-      : `tsh login --proxy=${host} --auth=local --user=${username}`;
+  let connectCmd;
+  switch (authType) {
+    case 'sso':
+      connectCmd = `tsh login --proxy=${host}`;
+      break;
+    case 'local':
+      connectCmd = `tsh login --proxy=${host} --auth=local --user=${username}`;
+      break;
+    case 'passwordless':
+      connectCmd = `tsh login --proxy=${host} --auth=passwordless --user=${username}`;
+      break;
+  }
```

Comment:
Thanks for the contribution!

As this is the same as what's used in `ConnectDialog.tsx`, could we make this into a helper function that lives in somewhere like `src/Databases/utils.ts`?

```suggestion
function getConnectCommand(authType: AuthType) {
  switch (authType) {
    case 'sso':
      return `tsh login --proxy=${host}`;
    case 'local':
      return `tsh login --proxy=${host} --auth=local --user=${username}`;
    case 'passwordless':
      return `tsh login --proxy=${host} --auth=passwordless --user=${username}`;
  }
  
  throw new Error(`auth type ${authType} is not supported`);
}
```

Should make it easier to add a test case for the logic 🙂

Reply:
@ryanclark
Good idea!
I couldn't find a `utils.ts` file in the `Databases` directory, but I located an existing function called `generateTshLoginCommand` in `packages/teleport/src/lib/util.ts`.
Do you think it would be appropriate to update the `generateTshLoginCommand` function to include passwordless auth, then we could reference `generateTshLoginCommand` from the database components and tests?

Reply:
Sorry for the confusion - I meant make `utils.ts` in the `Database` folder.

Good idea on changing `generateTshLoginCommand` to work the same - let's go for that, and it'll make testing easier as there's already tests for that method in https://github.com/gravitational/webapps/blob/master/packages/teleport/src/lib/util.test.ts#L17

Reply:
I've updated the `generateTshLoginCommand` function to check for the passwordless login authType, as well as refactored the `ConnectDialog` component to utilize it.

For the `AddDatabase` component however, I am unsure how to proceed since the `Props` type for the component does not include `clusterId` or `accessRequestId`, which are required for the `generateTshLoginCommand` function.

Could we just add those fields to the `Props` type of the `AddDatabase` component?

Reply:
You can make `clusterId` and `accessRequestId` optional on the `TshLoginCommand` type